### ISSUE TITLE
Using xinput mode instead of dinput

### ISSUE THREAD

**Issue by martindmtrv** [ahungry/vader3#3]
Hey @ahungry did you ever try to remap the controller with flydigi software then try and capture the keys in linux kernel? I wonder if we could do that then we don't need to use dinput mode. 

The main reason I want this is for steam deck OLED it seems the dinput mode cannot perform wake on bluetooth but xinput mode can 

Haven't had much chance to tinker with it lately but I'm curious about this

**Comment by ahungry** [comment]
Sadly I could not get that to be functional (custom key detection) via bluetooth+xinput.

Where I ended up:
- bluetooth + dinput = works great for my needs (I remap via qjoypad, as I tend to remap to keyboard binds, vs needing additional button binds), however lacks rumble
- bluetooth + xinput = lack of any real support for the special keys - I don't recall if I didn't see the events at all, or I only saw the remapped ones (when they were bound via the app, to other buttons - if you bind to keyboard events via the app, nothing will come through at all on GNU/Linux - thus the necessity to leave the flydigi app open in the background for that use case, even on Windows).

Beyond this driver (my xpad PR/update, to use with dongle):
- dinput = not functional at all
- xinput = working keybinds and working rumble, but requires using the dongle, which I find annoying on a laptop

If you figure something out, I'd love to hear though - 2 months in, and I'm still loving this controller!

**Comment by martindmtrv** [comment]
Yeah I may experiment with it more but if we could get Bluetooth + xinput with some remapping to match Xbox elite layout that would be the best!

Perhaps certain events could be passed through on the special buttons in xinput mode, since I saw a Reddit post float around that the macros defined in flydigi app would carry over to Linux

The setup would become a bit more janky though since it would depend on some flydigi configuration to work properly 

There may just be some way to get dinput controller recognized for wake from Bluetooth instead (it shows up as a mouse in the KDE Bluetooth settings instead of a controller)

**Comment by martindmtrv** [comment]
This controller is sweet though I agree! I also picked up a kk3 max to compare and the buttons and triggers are way better on the Vader

Remapping paddles on the fly (and without any software tricks) is nice though the kk3 has that going for it. But it does seem like the Vader 3 will always win out if the software can work correctly
